Submitted By: Martin Ward <macros_the_black@ntlworld.com>
Date: 01-02-2015
Initial Package Version: 4.9.2 
Origin: Idea originally developed by Ryan Oliver and Greg Schafer for
        the Pure LFS project. With contributions from William Harrington
Upstream Status: Not Applied - CLFS Specific
Description: The specs patch changes the location of the dynamic linker as
             well as the link spec to /opt/lib{,32,64}.

diff -Naur gcc-4.9.2.orig/gcc/config/aarch64/aarch64-linux.h gcc-4.9.2/gcc/config/aarch64/aarch64-linux.h
--- gcc-4.9.2.orig/gcc/config/aarch64/aarch64-linux.h	2015-02-01 08:25:12.425332500 +0000
+++ gcc-4.9.2/gcc/config/aarch64/aarch64-linux.h	2015-02-01 08:28:13.483028258 +0000
@@ -21,7 +21,7 @@
 #ifndef GCC_AARCH64_LINUX_H
 #define GCC_AARCH64_LINUX_H
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld-linux-aarch64%{mbig-endian:_be}.so.1"
+#define GLIBC_DYNAMIC_LINKER "/opt/lib/ld-linux-aarch64%{mbig-endian:_be}.so.1"
 
 #define CPP_SPEC "%{pthread:-D_REENTRANT}"
 
diff -Naur gcc-4.9.2.orig/gcc/config/alpha/linux-elf.h gcc-4.9.2/gcc/config/alpha/linux-elf.h
--- gcc-4.9.2.orig/gcc/config/alpha/linux-elf.h	2015-02-01 08:25:12.454332235 +0000
+++ gcc-4.9.2/gcc/config/alpha/linux-elf.h	2015-02-01 08:32:51.053547117 +0000
@@ -23,8 +23,8 @@
 #define EXTRA_SPECS \
 { "elf_dynamic_linker", ELF_DYNAMIC_LINKER },
 
-#define GLIBC_DYNAMIC_LINKER	"/lib/ld-linux.so.2"
-#define UCLIBC_DYNAMIC_LINKER "/lib/ld-uClibc.so.0"
+#define GLIBC_DYNAMIC_LINKER	"/opt/lib/ld-linux.so.2"
+#define UCLIBC_DYNAMIC_LINKER "/opt/lib/ld-uClibc.so.0"
 #if DEFAULT_LIBC == LIBC_UCLIBC
 #define CHOOSE_DYNAMIC_LINKER(G, U) "%{mglibc:" G ";:" U "}"
 #elif DEFAULT_LIBC == LIBC_GLIBC
diff -Naur gcc-4.9.2.orig/gcc/config/arm/linux-eabi.h gcc-4.9.2/gcc/config/arm/linux-eabi.h
--- gcc-4.9.2.orig/gcc/config/arm/linux-eabi.h	2015-02-01 08:25:12.426332491 +0000
+++ gcc-4.9.2/gcc/config/arm/linux-eabi.h	2015-02-01 08:34:09.315319673 +0000
@@ -68,8 +68,8 @@
    GLIBC_DYNAMIC_LINKER_DEFAULT and TARGET_DEFAULT_FLOAT_ABI.  */
 
 #undef  GLIBC_DYNAMIC_LINKER
-#define GLIBC_DYNAMIC_LINKER_SOFT_FLOAT "/lib/ld-linux.so.3"
-#define GLIBC_DYNAMIC_LINKER_HARD_FLOAT "/lib/ld-linux-armhf.so.3"
+#define GLIBC_DYNAMIC_LINKER_SOFT_FLOAT "/opt/lib/ld-linux.so.3"
+#define GLIBC_DYNAMIC_LINKER_HARD_FLOAT "/opt/lib/ld-linux-armhf.so.3"
 #define GLIBC_DYNAMIC_LINKER_DEFAULT GLIBC_DYNAMIC_LINKER_SOFT_FLOAT
 
 #define GLIBC_DYNAMIC_LINKER \
diff -Naur gcc-4.9.2.orig/gcc/config/arm/linux-elf.h gcc-4.9.2/gcc/config/arm/linux-elf.h
--- gcc-4.9.2.orig/gcc/config/arm/linux-elf.h	2015-02-01 08:25:12.430332454 +0000
+++ gcc-4.9.2/gcc/config/arm/linux-elf.h	2015-02-01 08:35:36.913990487 +0000
@@ -62,7 +62,7 @@
 
 #define LIBGCC_SPEC "%{mfloat-abi=soft*:-lfloat} -lgcc"
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER "/opt/lib/ld-linux.so.2"
 
 #define LINUX_TARGET_LINK_SPEC  "%{h*} \
    %{static:-Bstatic} \
diff -Naur gcc-4.9.2.orig/gcc/config/bfin/linux.h gcc-4.9.2/gcc/config/bfin/linux.h
--- gcc-4.9.2.orig/gcc/config/bfin/linux.h	2015-02-01 08:25:12.438332381 +0000
+++ gcc-4.9.2/gcc/config/bfin/linux.h	2015-02-01 08:36:15.209753338 +0000
@@ -45,7 +45,7 @@
   %{shared:-G -Bdynamic} \
   %{!shared: %{!static: \
    %{rdynamic:-export-dynamic} \
-   -dynamic-linker /lib/ld-uClibc.so.0} \
+   -dynamic-linker /opt/lib/ld-uClibc.so.0} \
    %{static}} -init __init -fini __fini"
 
 #undef TARGET_SUPPORTS_SYNC_CALLS
diff -Naur gcc-4.9.2.orig/gcc/config/cris/linux.h gcc-4.9.2/gcc/config/cris/linux.h
--- gcc-4.9.2.orig/gcc/config/cris/linux.h	2015-02-01 08:25:12.452332253 +0000
+++ gcc-4.9.2/gcc/config/cris/linux.h	2015-02-01 08:36:59.889437277 +0000
@@ -102,7 +102,7 @@
 #undef CRIS_DEFAULT_CPU_VERSION
 #define CRIS_DEFAULT_CPU_VERSION CRIS_CPU_NG
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER "/opt/lib/ld.so.1"
 
 #undef CRIS_LINK_SUBTARGET_SPEC
 #define CRIS_LINK_SUBTARGET_SPEC \
diff -Naur gcc-4.9.2.orig/gcc/config/freebsd-spec.h gcc-4.9.2/gcc/config/freebsd-spec.h
--- gcc-4.9.2.orig/gcc/config/freebsd-spec.h	2015-02-01 08:25:12.421332536 +0000
+++ gcc-4.9.2/gcc/config/freebsd-spec.h	2015-02-01 08:39:22.743148494 +0000
@@ -128,9 +128,9 @@
 #endif
 
 #if FBSD_MAJOR < 6
-#define FBSD_DYNAMIC_LINKER "/usr/libexec/ld-elf.so.1"
+#define FBSD_DYNAMIC_LINKER "/opt/libexec/ld-elf.so.1"
 #else
-#define FBSD_DYNAMIC_LINKER "/libexec/ld-elf.so.1"
+#define FBSD_DYNAMIC_LINKER "/opt/libexec/ld-elf.so.1"
 #endif
 
 /* NOTE: The freebsd-spec.h header is included also for various
diff -Naur gcc-4.9.2.orig/gcc/config/frv/linux.h gcc-4.9.2/gcc/config/frv/linux.h
--- gcc-4.9.2.orig/gcc/config/frv/linux.h	2015-02-01 08:25:12.455332226 +0000
+++ gcc-4.9.2/gcc/config/frv/linux.h	2015-02-01 08:40:30.870426618 +0000
@@ -34,7 +34,7 @@
 #define ENDFILE_SPEC \
   "%{shared|pie:crtendS.o%s;:crtend.o%s} crtn.o%s"
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER "/opt/lib/ld.so.1"
 
 #undef LINK_SPEC
 #define LINK_SPEC "\
diff -Naur gcc-4.9.2.orig/gcc/config/i386/gnu.h gcc-4.9.2/gcc/config/i386/gnu.h
--- gcc-4.9.2.orig/gcc/config/i386/gnu.h	2015-02-01 08:25:12.445332317 +0000
+++ gcc-4.9.2/gcc/config/i386/gnu.h	2015-02-01 08:41:24.733821511 +0000
@@ -22,7 +22,7 @@
 #define GNU_USER_LINK_EMULATION "elf_i386"
 
 #undef GNU_USER_DYNAMIC_LINKER
-#define GNU_USER_DYNAMIC_LINKER "/lib/ld.so"
+#define GNU_USER_DYNAMIC_LINKER "/opt/lib/ld.so"
 
 #undef	STARTFILE_SPEC
 #if defined HAVE_LD_PIE
diff -Naur gcc-4.9.2.orig/gcc/config/i386/kfreebsd-gnu64.h gcc-4.9.2/gcc/config/i386/kfreebsd-gnu64.h
--- gcc-4.9.2.orig/gcc/config/i386/kfreebsd-gnu64.h	2015-02-01 08:25:12.446332308 +0000
+++ gcc-4.9.2/gcc/config/i386/kfreebsd-gnu64.h	2015-02-01 08:45:19.329958472 +0000
@@ -22,6 +22,6 @@
 #define GNU_USER_LINK_EMULATION64 "elf_x86_64_fbsd"
 #define GNU_USER_LINK_EMULATIONX32 "elf32_x86_64_fbsd"
 
-#define GLIBC_DYNAMIC_LINKER32 "/lib/ld.so.1"
-#define GLIBC_DYNAMIC_LINKER64 "/lib/ld-kfreebsd-x86-64.so.1"
-#define GLIBC_DYNAMIC_LINKERX32 "/lib/ld-kfreebsd-x32.so.1"
+#define GLIBC_DYNAMIC_LINKER32 "/opt/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER64 "/opt/lib/ld-kfreebsd-x86-64.so.1"
+#define GLIBC_DYNAMIC_LINKERX32 "/opt/lib/ld-kfreebsd-x32.so.1"
diff -Naur gcc-4.9.2.orig/gcc/config/i386/kfreebsd-gnu.h gcc-4.9.2/gcc/config/i386/kfreebsd-gnu.h
--- gcc-4.9.2.orig/gcc/config/i386/kfreebsd-gnu.h	2015-02-01 08:25:12.445332317 +0000
+++ gcc-4.9.2/gcc/config/i386/kfreebsd-gnu.h	2015-02-01 08:44:28.403602862 +0000
@@ -19,4 +19,4 @@
 <http://www.gnu.org/licenses/>.  */
 
 #define GNU_USER_LINK_EMULATION "elf_i386_fbsd"
-#define GLIBC_DYNAMIC_LINKER "/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER "/opt/lib/ld.so.1"
diff -Naur gcc-4.9.2.orig/gcc/config/i386/linux64.h gcc-4.9.2/gcc/config/i386/linux64.h
--- gcc-4.9.2.orig/gcc/config/i386/linux64.h	2015-02-01 08:25:12.445332317 +0000
+++ gcc-4.9.2/gcc/config/i386/linux64.h	2015-02-01 08:42:39.988936800 +0000
@@ -27,6 +27,6 @@
 #define GNU_USER_LINK_EMULATION64 "elf_x86_64"
 #define GNU_USER_LINK_EMULATIONX32 "elf32_x86_64"
 
-#define GLIBC_DYNAMIC_LINKER32 "/lib/ld-linux.so.2"
-#define GLIBC_DYNAMIC_LINKER64 "/lib64/ld-linux-x86-64.so.2"
-#define GLIBC_DYNAMIC_LINKERX32 "/libx32/ld-linux-x32.so.2"
+#define GLIBC_DYNAMIC_LINKER32 "/opt/lib/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER64 "/opt/lib64/ld-linux-x86-64.so.2"
+#define GLIBC_DYNAMIC_LINKERX32 "/opt/libx32/ld-linux-x32.so.2"
diff -Naur gcc-4.9.2.orig/gcc/config/i386/linux.h gcc-4.9.2/gcc/config/i386/linux.h
--- gcc-4.9.2.orig/gcc/config/i386/linux.h	2015-02-01 08:25:12.446332308 +0000
+++ gcc-4.9.2/gcc/config/i386/linux.h	2015-02-01 08:41:45.669579546 +0000
@@ -20,4 +20,4 @@
 <http://www.gnu.org/licenses/>.  */
 
 #define GNU_USER_LINK_EMULATION "elf_i386"
-#define GLIBC_DYNAMIC_LINKER "/lib/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER "/opt/lib/ld-linux.so.2"
diff -Naur gcc-4.9.2.orig/gcc/config/ia64/linux.h gcc-4.9.2/gcc/config/ia64/linux.h
--- gcc-4.9.2.orig/gcc/config/ia64/linux.h	2015-02-01 08:25:12.467332116 +0000
+++ gcc-4.9.2/gcc/config/ia64/linux.h	2015-02-01 08:46:02.521405159 +0000
@@ -55,7 +55,7 @@
 /* Define this for shared library support because it isn't in the main
    linux.h file.  */
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld-linux-ia64.so.2"
+#define GLIBC_DYNAMIC_LINKER "/opt/lib/ld-linux-ia64.so.2"
 
 #undef LINK_SPEC
 #define LINK_SPEC "\
diff -Naur gcc-4.9.2.orig/gcc/config/knetbsd-gnu.h gcc-4.9.2/gcc/config/knetbsd-gnu.h
--- gcc-4.9.2.orig/gcc/config/knetbsd-gnu.h	2015-02-01 08:25:12.420332546 +0000
+++ gcc-4.9.2/gcc/config/knetbsd-gnu.h	2015-02-01 08:46:28.769066300 +0000
@@ -32,4 +32,4 @@
 
 
 #undef GNU_USER_DYNAMIC_LINKER
-#define GNU_USER_DYNAMIC_LINKER "/lib/ld.so.1"
+#define GNU_USER_DYNAMIC_LINKER "/opt/lib/ld.so.1"
diff -Naur gcc-4.9.2.orig/gcc/config/kopensolaris-gnu.h gcc-4.9.2/gcc/config/kopensolaris-gnu.h
--- gcc-4.9.2.orig/gcc/config/kopensolaris-gnu.h	2015-02-01 08:25:12.422332527 +0000
+++ gcc-4.9.2/gcc/config/kopensolaris-gnu.h	2015-02-01 08:47:02.272631276 +0000
@@ -31,4 +31,5 @@
   while (0)
 
 #undef GNU_USER_DYNAMIC_LINKER
-#define GNU_USER_DYNAMIC_LINKER "/lib/ld.so.1"
+#define GNU_USER_DYNAMIC_LINKER "/opt/lib/ld.so.1"
+
diff -Naur gcc-4.9.2.orig/gcc/config/linux.h gcc-4.9.2/gcc/config/linux.h
--- gcc-4.9.2.orig/gcc/config/linux.h	2015-02-01 08:25:12.421332536 +0000
+++ gcc-4.9.2/gcc/config/linux.h	2015-02-01 08:48:50.191214604 +0000
@@ -73,10 +73,10 @@
    GLIBC_DYNAMIC_LINKER must be defined for each target using them, or
    GLIBC_DYNAMIC_LINKER32 and GLIBC_DYNAMIC_LINKER64 for targets
    supporting both 32-bit and 64-bit compilation.  */
-#define UCLIBC_DYNAMIC_LINKER "/lib/ld-uClibc.so.0"
-#define UCLIBC_DYNAMIC_LINKER32 "/lib/ld-uClibc.so.0"
-#define UCLIBC_DYNAMIC_LINKER64 "/lib/ld64-uClibc.so.0"
-#define UCLIBC_DYNAMIC_LINKERX32 "/lib/ldx32-uClibc.so.0"
+#define UCLIBC_DYNAMIC_LINKER "/opt/lib/ld-uClibc.so.0"
+#define UCLIBC_DYNAMIC_LINKER32 "/opt/lib/ld-uClibc.so.0"
+#define UCLIBC_DYNAMIC_LINKER64 "/opt/lib/ld64-uClibc.so.0"
+#define UCLIBC_DYNAMIC_LINKERX32 "/opt/lib/ldx32-uClibc.so.0"
 #define BIONIC_DYNAMIC_LINKER "/system/bin/linker"
 #define BIONIC_DYNAMIC_LINKER32 "/system/bin/linker"
 #define BIONIC_DYNAMIC_LINKER64 "/system/bin/linker64"
diff -Naur gcc-4.9.2.orig/gcc/config/lm32/uclinux-elf.h gcc-4.9.2/gcc/config/lm32/uclinux-elf.h
--- gcc-4.9.2.orig/gcc/config/lm32/uclinux-elf.h	2015-02-01 08:25:12.437332390 +0000
+++ gcc-4.9.2/gcc/config/lm32/uclinux-elf.h	2015-02-01 08:50:03.637240283 +0000
@@ -67,7 +67,7 @@
    %{shared:-shared} \
    %{symbolic:-Bsymbolic} \
    %{rdynamic:-export-dynamic} \
-   -dynamic-linker /lib/ld-linux.so.2"
+   -dynamic-linker /opt/lib/ld-linux.so.2"
 
 #define TARGET_OS_CPP_BUILTINS() GNU_USER_TARGET_OS_CPP_BUILTINS()
 
diff -Naur gcc-4.9.2.orig/gcc/config/m68k/linux.h gcc-4.9.2/gcc/config/m68k/linux.h
--- gcc-4.9.2.orig/gcc/config/m68k/linux.h	2015-02-01 08:25:12.437332390 +0000
+++ gcc-4.9.2/gcc/config/m68k/linux.h	2015-02-01 08:50:44.764691988 +0000
@@ -71,7 +71,7 @@
    When the -shared link option is used a final link is not being
    done.  */
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER "/opt/lib/ld.so.1"
 
 #undef LINK_SPEC
 #define LINK_SPEC "-m m68kelf %{shared} \
diff -Naur gcc-4.9.2.orig/gcc/config/microblaze/linux.h gcc-4.9.2/gcc/config/microblaze/linux.h
--- gcc-4.9.2.orig/gcc/config/microblaze/linux.h	2015-02-01 08:25:12.457332207 +0000
+++ gcc-4.9.2/gcc/config/microblaze/linux.h	2015-02-01 08:51:12.052327338 +0000
@@ -25,7 +25,7 @@
 #undef TLS_NEEDS_GOT
 #define TLS_NEEDS_GOT 1
 
-#define DYNAMIC_LINKER "/lib/ld.so.1"
+#define DYNAMIC_LINKER "/opt/lib/ld.so.1"
 #undef  SUBTARGET_EXTRA_SPECS
 #define SUBTARGET_EXTRA_SPECS \
   { "dynamic_linker", DYNAMIC_LINKER }
diff -Naur gcc-4.9.2.orig/gcc/config/mips/linux64.h gcc-4.9.2/gcc/config/mips/linux64.h
--- gcc-4.9.2.orig/gcc/config/mips/linux64.h	2015-02-01 08:25:12.432332436 +0000
+++ gcc-4.9.2/gcc/config/mips/linux64.h	2015-02-01 08:53:59.896767713 +0000
@@ -23,20 +23,20 @@
 #define GNU_USER_LINK_EMULATIONN32 "elf32%{EB:b}%{EL:l}tsmipn32"
 
 #define GLIBC_DYNAMIC_LINKER32 \
-  "%{mnan=2008:/lib/ld-linux-mipsn8.so.1;:/lib/ld.so.1}"
+  "%{mnan=2008:/opt/lib/ld-linux-mipsn8.so.1;:/opt/lib/ld.so.1}"
 #define GLIBC_DYNAMIC_LINKER64 \
-  "%{mnan=2008:/lib64/ld-linux-mipsn8.so.1;:/lib64/ld.so.1}"
+  "%{mnan=2008:/opt/lib64/ld-linux-mipsn8.so.1;:/opt/lib64/ld.so.1}"
 #define GLIBC_DYNAMIC_LINKERN32 \
-  "%{mnan=2008:/lib32/ld-linux-mipsn8.so.1;:/lib32/ld.so.1}"
+  "%{mnan=2008:/opt/lib32/ld-linux-mipsn8.so.1;:/opt/lib32/ld.so.1}"
 
 #undef UCLIBC_DYNAMIC_LINKER32
 #define UCLIBC_DYNAMIC_LINKER32 \
-  "%{mnan=2008:/lib/ld-uClibc-mipsn8.so.0;:/lib/ld-uClibc.so.0}"
+  "%{mnan=2008:/opt/lib/ld-uClibc-mipsn8.so.0;:/opt/lib/ld-uClibc.so.0}"
 #undef UCLIBC_DYNAMIC_LINKER64
 #define UCLIBC_DYNAMIC_LINKER64 \
-  "%{mnan=2008:/lib/ld64-uClibc-mipsn8.so.0;:/lib/ld64-uClibc.so.0}"
+  "%{mnan=2008:/opt/lib/ld64-uClibc-mipsn8.so.0;:/opt/lib/ld64-uClibc.so.0}"
 #define UCLIBC_DYNAMIC_LINKERN32 \
-  "%{mnan=2008:/lib32/ld-uClibc-mipsn8.so.0;:/lib32/ld-uClibc.so.0}"
+  "%{mnan=2008:/opt/lib32/ld-uClibc-mipsn8.so.0;:/opt/lib32/ld-uClibc.so.0}"
 
 #define BIONIC_DYNAMIC_LINKERN32 "/system/bin/linker32"
 #define GNU_USER_DYNAMIC_LINKERN32 \
diff -Naur gcc-4.9.2.orig/gcc/config/mips/linux.h gcc-4.9.2/gcc/config/mips/linux.h
--- gcc-4.9.2.orig/gcc/config/mips/linux.h	2015-02-01 08:25:12.433332427 +0000
+++ gcc-4.9.2/gcc/config/mips/linux.h	2015-02-01 08:52:15.187481366 +0000
@@ -18,8 +18,8 @@
 <http://www.gnu.org/licenses/>.  */
 
 #define GLIBC_DYNAMIC_LINKER \
-  "%{mnan=2008:/lib/ld-linux-mipsn8.so.1;:/lib/ld.so.1}"
+  "%{mnan=2008:/opt/lib/ld-linux-mipsn8.so.1;:/opt/lib/ld.so.1}"
 
 #undef UCLIBC_DYNAMIC_LINKER
 #define UCLIBC_DYNAMIC_LINKER \
-  "%{mnan=2008:/lib/ld-uClibc-mipsn8.so.0;:/lib/ld-uClibc.so.0}"
+  "%{mnan=2008:/opt/lib/ld-uClibc-mipsn8.so.0;:/opt/lib/ld-uClibc.so.0}"
diff -Naur gcc-4.9.2.orig/gcc/config/mn10300/linux.h gcc-4.9.2/gcc/config/mn10300/linux.h
--- gcc-4.9.2.orig/gcc/config/mn10300/linux.h	2015-02-01 08:25:12.458332198 +0000
+++ gcc-4.9.2/gcc/config/mn10300/linux.h	2015-02-01 08:54:48.823936710 +0000
@@ -32,7 +32,7 @@
 #undef  ASM_SPEC
 #define ASM_SPEC ""
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER "/opt/lib/ld.so.1"
 
 #undef  LINK_SPEC
 #define LINK_SPEC "%{mrelax:--relax} %{shared:-shared} \
diff -Naur gcc-4.9.2.orig/gcc/config/pa/pa-linux.h gcc-4.9.2/gcc/config/pa/pa-linux.h
--- gcc-4.9.2.orig/gcc/config/pa/pa-linux.h	2015-02-01 08:25:12.458332198 +0000
+++ gcc-4.9.2/gcc/config/pa/pa-linux.h	2015-02-01 08:55:21.879382779 +0000
@@ -40,7 +40,7 @@
 /* Define this for shared library support because it isn't in the main
    linux.h file.  */
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER "/opt/lib/ld.so.1"
 
 #undef LINK_SPEC
 #define LINK_SPEC "\
diff -Naur gcc-4.9.2.orig/gcc/config/rs6000/linux64.h gcc-4.9.2/gcc/config/rs6000/linux64.h
--- gcc-4.9.2.orig/gcc/config/rs6000/linux64.h	2015-02-01 08:25:12.462332162 +0000
+++ gcc-4.9.2/gcc/config/rs6000/linux64.h	2015-02-01 08:59:41.962195144 +0000
@@ -367,14 +367,14 @@
 #undef	LINK_OS_DEFAULT_SPEC
 #define LINK_OS_DEFAULT_SPEC "%(link_os_linux)"
 
-#define GLIBC_DYNAMIC_LINKER32 "/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER32 "/opt/lib/ld.so.1"
 #ifdef LINUX64_DEFAULT_ABI_ELFv2
-#define GLIBC_DYNAMIC_LINKER64 "%{mabi=elfv1:/lib64/ld64.so.1;:/lib64/ld64.so.2}"
+#define GLIBC_DYNAMIC_LINKER64 "%{mabi=elfv1:/opt/lib64/ld64.so.1;:/opt/lib64/ld64.so.2}"
 #else
-#define GLIBC_DYNAMIC_LINKER64 "%{mabi=elfv2:/lib64/ld64.so.2;:/lib64/ld64.so.1}"
+#define GLIBC_DYNAMIC_LINKER64 "%{mabi=elfv2:/opt/lib64/ld64.so.2;:/opt/lib64/ld64.so.1}"
 #endif
-#define UCLIBC_DYNAMIC_LINKER32 "/lib/ld-uClibc.so.0"
-#define UCLIBC_DYNAMIC_LINKER64 "/lib/ld64-uClibc.so.0"
+#define UCLIBC_DYNAMIC_LINKER32 "/opt/lib/ld-uClibc.so.0"
+#define UCLIBC_DYNAMIC_LINKER64 "/opt/lib/ld64-uClibc.so.0"
 #if DEFAULT_LIBC == LIBC_UCLIBC
 #define CHOOSE_DYNAMIC_LINKER(G, U) "%{mglibc:" G ";:" U "}"
 #elif DEFAULT_LIBC == LIBC_GLIBC
diff -Naur gcc-4.9.2.orig/gcc/config/rs6000/sysv4.h gcc-4.9.2/gcc/config/rs6000/sysv4.h
--- gcc-4.9.2.orig/gcc/config/rs6000/sysv4.h	2015-02-01 08:25:12.463332153 +0000
+++ gcc-4.9.2/gcc/config/rs6000/sysv4.h	2015-02-01 09:02:34.127546976 +0000
@@ -761,8 +761,8 @@
 
 #define LINK_START_LINUX_SPEC ""
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld.so.1"
-#define UCLIBC_DYNAMIC_LINKER "/lib/ld-uClibc.so.0"
+#define GLIBC_DYNAMIC_LINKER "/opt/lib/ld.so.1"
+#define UCLIBC_DYNAMIC_LINKER "/opt/lib/ld-uClibc.so.0"
 #if DEFAULT_LIBC == LIBC_UCLIBC
 #define CHOOSE_DYNAMIC_LINKER(G, U) "%{mglibc:" G ";:" U "}"
 #elif !defined (DEFAULT_LIBC) || DEFAULT_LIBC == LIBC_GLIBC
diff -Naur gcc-4.9.2.orig/gcc/config/s390/linux.h gcc-4.9.2/gcc/config/s390/linux.h
--- gcc-4.9.2.orig/gcc/config/s390/linux.h	2015-02-01 08:25:12.443332335 +0000
+++ gcc-4.9.2/gcc/config/s390/linux.h	2015-02-01 09:03:44.190489439 +0000
@@ -60,8 +60,8 @@
 #define MULTILIB_DEFAULTS { "m31" }
 #endif
 
-#define GLIBC_DYNAMIC_LINKER32 "/lib/ld.so.1"
-#define GLIBC_DYNAMIC_LINKER64 "/lib/ld64.so.1"
+#define GLIBC_DYNAMIC_LINKER32 "/opt/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER64 "/opt/lib/ld64.so.1"
 
 #undef  LINK_SPEC
 #define LINK_SPEC \
diff -Naur gcc-4.9.2.orig/gcc/config/sh/linux.h gcc-4.9.2/gcc/config/sh/linux.h
--- gcc-4.9.2.orig/gcc/config/sh/linux.h	2015-02-01 08:25:12.465332134 +0000
+++ gcc-4.9.2/gcc/config/sh/linux.h	2015-02-01 09:04:25.372872341 +0000
@@ -43,7 +43,7 @@
 
 #define TARGET_ASM_FILE_END file_end_indicate_exec_stack
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER "/opt/lib/ld-linux.so.2"
 
 #undef SUBTARGET_LINK_EMUL_SUFFIX
 #define SUBTARGET_LINK_EMUL_SUFFIX "_linux"
diff -Naur gcc-4.9.2.orig/gcc/config/sparc/linux64.h gcc-4.9.2/gcc/config/sparc/linux64.h
--- gcc-4.9.2.orig/gcc/config/sparc/linux64.h	2015-02-01 08:25:12.456332216 +0000
+++ gcc-4.9.2/gcc/config/sparc/linux64.h	2015-02-01 09:06:56.362633950 +0000
@@ -92,8 +92,8 @@
    When the -shared link option is used a final link is not being
    done.  */
 
-#define GLIBC_DYNAMIC_LINKER32 "/lib/ld-linux.so.2"
-#define GLIBC_DYNAMIC_LINKER64 "/lib64/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER32 "/opt/lib/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER64 "/opt/lib64/ld-linux.so.2"
 
 #ifdef SPARC_BI_ARCH
 
@@ -201,7 +201,7 @@
 #else /* !SPARC_BI_ARCH */
 
 #undef LINK_SPEC
-#define LINK_SPEC "-m elf64_sparc -Y P,%R/usr/lib64 %{shared:-shared} \
+#define LINK_SPEC "-m elf64_sparc -Y P,%R/opt/lib64 %{shared:-shared} \
   %{!shared: \
     %{!static: \
       %{rdynamic:-export-dynamic} \
diff -Naur gcc-4.9.2.orig/gcc/config/sparc/linux.h gcc-4.9.2/gcc/config/sparc/linux.h
--- gcc-4.9.2.orig/gcc/config/sparc/linux.h	2015-02-01 08:25:12.456332216 +0000
+++ gcc-4.9.2/gcc/config/sparc/linux.h	2015-02-01 09:05:04.980281698 +0000
@@ -83,7 +83,7 @@
    When the -shared link option is used a final link is not being
    done.  */
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER "/opt/lib/ld-linux.so.2"
 
 #undef  LINK_SPEC
 #define LINK_SPEC "-m elf32_sparc %{shared:-shared} \
diff -Naur gcc-4.9.2.orig/gcc/config/vax/linux.h gcc-4.9.2/gcc/config/vax/linux.h
--- gcc-4.9.2.orig/gcc/config/vax/linux.h	2015-02-01 08:25:12.457332207 +0000
+++ gcc-4.9.2/gcc/config/vax/linux.h	2015-02-01 09:07:35.797054699 +0000
@@ -41,7 +41,7 @@
   %{!shared: \
     %{!static: \
       %{rdynamic:-export-dynamic} \
-      -dynamic-linker /lib/ld.so.1} \
+      -dynamic-linker /opt/lib/ld.so.1} \
     %{static:-static}}"
 
 #undef  WCHAR_TYPE
diff -Naur gcc-4.9.2.orig/gcc/config/xtensa/linux.h gcc-4.9.2/gcc/config/xtensa/linux.h
--- gcc-4.9.2.orig/gcc/config/xtensa/linux.h	2015-02-01 08:25:12.422332527 +0000
+++ gcc-4.9.2/gcc/config/xtensa/linux.h	2015-02-01 09:08:06.601603551 +0000
@@ -44,7 +44,7 @@
   %{mlongcalls:--longcalls} \
   %{mno-longcalls:--no-longcalls}"
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER "/opt/lib/ld.so.1"
 
 #undef LINK_SPEC
 #define LINK_SPEC \
